tests/binning_test.py

```diff
@@ -547,3 +547,60 @@ def test_variable_bin_equivalent_to_bin_of_data_array_with_counts():
     coord = sc.array(dims=['row'], values=[2, 2, 1, 3])
     da = sc.DataArray(data, coords={'x': coord})
     assert sc.identical(coord.bin(x=3), da.bin(x=3))
+
+
+def test_group_many_to_many_uses_optimized_codepath():
+    size = 512
+    table = sc.data.table_xyz(int(1e3))
+    da = table.bin(x=size, y=size)
+    xcoarse = sc.arange('x', size) // 2
+    ycoarse = sc.arange('y', size) // 2
+    da.coords['xcoarse'] = xcoarse
+    da.coords['ycoarse'] = ycoarse
+    # The old "standard" implementation uses a massive amount of memory, crashing even
+    # with 256 GByte of RAM.
```

Comment:
These tests are a bit odd. I agree that we should test that commonly used operations work. But it depends on the resources of the test runner. So this might fail if the test runs on old hardware or the old codepath might work in the future.

Reply:
I don't think it would fail on old hardware. The old codepath would take many minutes, so I think we would notice? In general I agree, but do you have a better idea?

Reply:
Is there a way of identifying that the correct path was used apart from the fact that it didn't crash?
A telling sign in the output?... probably not is my guess

Reply:
> A telling sign in the output?

That would indicate a bug, since we want to guarantee the same behavior in all code paths.